server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -0,0 +1,284 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE)
+public class SearchBackpressureIT extends OpenSearchIntegTestCase {
+
+    private static final TimeValue TIMEOUT = new TimeValue(30, TimeUnit.SECONDS);
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(SearchBackpressureSettings.SETTING_CANCELLATION_RATIO.getKey(), 1)
+            .put(SearchBackpressureSettings.SETTING_CANCELLATION_RATE.getKey(), 10)
+            .put(SearchBackpressureSettings.SETTING_CANCELLATION_BURST.getKey(), 10)
+            .put(NodeDuressSettings.SETTING_CPU_THRESHOLD.getKey(), 0.3)
+            .put(NodeDuressSettings.SETTING_HEAP_THRESHOLD.getKey(), 0.0)
+            .put(NodeDuressSettings.SETTING_NUM_SUCCESSIVE_BREACHES.getKey(), 1)
+            .put(SearchShardTaskSettings.SETTING_TOTAL_HEAP_PERCENT_THRESHOLD.getKey(), 0.0)
+            .build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        final List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(TestPlugin.class);
+        return plugins;
+    }
+
+    public void testSearchShardTaskCancellationWithHighElapsedTime() throws InterruptedException {
+        ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.persistentSettings(
+            Settings.builder().put(ElapsedTimeTracker.SETTING_ELAPSED_TIME_MILLIS_THRESHOLD.getKey(), 5000)
+        );
+        assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, generateRequestWithHighElapsedTime(8000), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                fail("SearchShardTask should have been cancelled");
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.info("testSearchShardTaskCancellationWithHighElapsedTime onFailure e = " + e.getMessage());
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("elapsed time exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await();
```

Comment:
Could we please bound the await time to, may be, 10 seconds?

Reply:
Added timeout of 30 seconds, same as the TestRequest timeout defined in the test.